File: LeetCode/p1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = ""
        for a, b in zip(word1, word2):
            res += a + b

        if len(word1) > len(word2):
            res += word1[len(word2):]
        elif len(word2) > len(word1):
            res += word2[len(word1):]

        return res


a = Solution().mergeAlternately("abcdef", "pqr")
print(a)

File: LeetCode-Java/p1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder res = new StringBuilder();
        int i = 0;
        
        // Merge characters alternately until the shorter string ends
        while (i < word1.length() && i < word2.length()) {
            res.append(word1.charAt(i));
            res.append(word2.charAt(i));
            i++;
        }
        
        // Append remaining characters from word1 if any
        if (i < word1.length()) {
            res.append(word1.substring(i));
        }
        
        // Append remaining characters from word2 if any
        if (i < word2.length()) {
            res.append(word2.substring(i));
        }
        
        return res.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String result = solution.mergeAlternately("abcdef", "pqr");
        System.out.println(result);
    }
}